### PR TITLE
feat: shell and composer cosmetics

### PR DESCRIPTION
**Pull request by zhenyasav**
https://github.com/dxos/dxos/assets/3398896/6f50121e-9c61-44d1-ba76-bc53115fe1f5

### Summary
📚🌈🎛️

This pull request adds Storybook support and a ProgressBar component to the `composer-app` package, and updates some UI components in the `react-shell` package to use the `Action` component and the `aurora-theme` variables. It also modifies some style variables and animations in the `aurora-theme` package, and changes the vault.html file to use the dark theme.

> _We added some stories and styles_
> _To make our UI components beguile_
> _We used `Action` and `ProgressBar`_
> _And tweaked some colors and props so far_
> _And tested them all with Storybook's dial_

### Walkthrough
*  Add Storybook configuration and stories for the composer-app package ([link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-aa91966b768fadbc4461b4ea044e46a0df9ab8291b1803ce7bd261890de2e1b0R1-R32), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-f46586193d8c170c18f0c20f3cfe6bfbbc1fab6bbe260893ecc54ed7c156284aR1-R3), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-8369236928d3b3a548e656007587690c4996aa87f37f281489b92372715f846aR1-R50), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-305064e5650658acc533706323113181b31fa246d06b6d28dbb047dd44f8578cL79-R90), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-2afb21850059be1c1ec9239b95be9e965d32cc03f8d68435acc0c1cbc10a8ab3R1-R32), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-87f34605e837584dfb4134e5618b3e3803c2ac7b8e2a20f139195b8ab1d55df8L1-L-1))
*  Implement and use the ProgressBar component to show loading indicators in the composer-app package ([link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-953d7ab7fa5ee815d97020ae5ef3a5314de864f657cd91d43d9bc8b27de08a36R1-R29), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-9f2aee287f92edd662c13bed460a28a1590c9186b62c98ccc2e00f6877c3d7e6R30-R31), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-9f2aee287f92edd662c13bed460a28a1590c9186b62c98ccc2e00f6877c3d7e6L39-R45), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-3af43445f59986a74a4351ba74ecd39e2125c7ae0194284ac6374e133a33939eR45), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-104ae02f9b930ab0f54259fab127bc58060e28f0c63fc48f7a52913ba4626acaR307-R320), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-104ae02f9b930ab0f54259fab127bc58060e28f0c63fc48f7a52913ba4626acaR337-R338))
*  Replace the Button component with the Action component in the react-shell package, and add a variant prop to the Action component ([link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-7fe4da9577079692c74c85e436e71ecc6b1d19ece2a80b4c0ec5dde93d832280L14-R35), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-66cf64005170336917b17ad61aef44a1853a2242f4aa49cc1af821756c945058L7-R12), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-e5d2b5109114f04563cde5dfec2a39e6e641c51b7ff684736f6d13b10fd4b90fR1-R47), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-fa8c68d7610e9d6cbb557f18e549d82b80c1c81dd5fee31b2b45d0260bd34e8fL9-R9), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-fa8c68d7610e9d6cbb557f18e549d82b80c1c81dd5fee31b2b45d0260bd34e8fL22-R22), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-fa8c68d7610e9d6cbb557f18e549d82b80c1c81dd5fee31b2b45d0260bd34e8fL28-R28), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-fa8c68d7610e9d6cbb557f18e549d82b80c1c81dd5fee31b2b45d0260bd34e8fL34-R34), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-83479971032890b491c8483fffed89b95e93f4800dc3040839d3d9d2454552abR15), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-83479971032890b491c8483fffed89b95e93f4800dc3040839d3d9d2454552abL62-R63), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-83479971032890b491c8483fffed89b95e93f4800dc3040839d3d9d2454552abL71-R78), [link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-83479971032890b491c8483fffed89b95e93f4800dc3040839d3d9d2454552abL86-R87))
*  Add a cutout prop to the InvitationManagerView component in the react-shell package, to create a different background and border style ([link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-3af43445f59986a74a4351ba74ecd39e2125c7ae0194284ac6374e133a33939eR86))
*  Force the dark theme in the vault.html file in the vault package, for testing purposes ([link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-c6591874d4c068ce9808c13cf79c5100b581bdefc219b26f84e222519ef770e9L37-R37))
*  Modify the groupSurface variable in the surface.ts file in the aurora-theme package, to add transparency, blur, and dark mode ([link](https://github.com/dxos/dxos/pull/4087/files?diff=unified&w=0#diff-376d38daf5becffcdfe06f83aa50e5963917345596655316d091c4fa298dcb40L12-R12))

### CODE REVIEW

File: packages/apps/plugins/plugin-markdown/src/components/StandaloneLayout.tsx [hunk 1]
```diff
@@ -24,8 +24,8 @@ export const StandaloneLayout = ({
         <div
           role='none'
           className={mx(
-            inputSurface,
-            surfaceElevation({ elevation: 'group' }),
+            // inputSurface,
+            // surfaceElevation({ elevation: 'group' }),
```
Comment (thure) [inline packages/apps/plugins/plugin-markdown/src/components/StandaloneLayout.tsx:28]: Rich and I added these to harmonize MarkdownPlugin’s main surface with other plugins’ main surfaces. I think making experiences that are equivalent also look equivalent and look like each other is a worthwhile design hypothesis. I'd prefer to continue exposing folks to that idea before going back to a treatment just for MarkdownPlugin that is flat & transparent like the other static elements in the UI.

**Review by thure — CHANGES_REQUESTED** [review]
This is looking good! Lots of worthwhile ideas to try, and I'm very glad this introduces a Storybook for Composer and other plugins!

I notice an interesting non-flush chrome surface treatment on the topbar’s child elements. Is this why you flattened MarkdownPlugin’s main area?

I think I'd prefer to take a path that doesn’t trade the main area’s usability / expressive range for the topbar’s visual appeal, which is the only reason I'm requesting changes. I'm looking forward to trying out all the other ideas here!

File: packages/apps/plugins/plugin-graph/src/graph/types.ts [hunk 2]
```diff
@@ -99,7 +99,7 @@ export namespace Graph {
      * @example ['unknown node label, { ns: 'example-plugin' }]
      */
     // TODO(thure): `Parameters<TFunction>` causes typechecking issues because `TFunction` has so many signatures.
-    label: string | [string, { ns: string; count?: number }];
+    label?: string | [string, { ns: string; count?: number }];
```
Comment (wittjosiah) [inline packages/apps/plugins/plugin-graph/src/graph/types.ts:102]: Why optional?

File: packages/sdk/vault/vault.html [hunk 3]
```diff
@@ -34,7 +34,7 @@
         document.documentElement.classList[darkMode ? 'add' : 'remove']('dark')
       }
       const paletteMatch = window.matchMedia('(prefers-color-scheme: dark)');
-      if (window.top === window) {
+      if (true || window.top === window) {
```
Comment (wittjosiah) [inline packages/sdk/vault/vault.html:37]: :eyes:

File: packages/apps/plugins/plugin-markdown/src/components/StandaloneLayout.tsx (hunk 1 above)
Comment (zhenyasav) [reply packages/apps/plugins/plugin-markdown/src/components/StandaloneLayout.tsx:28]: not sure which other plugins we should care about here, but i'm pretty convinced we don't want another frame within-a-frame when we're working on a document, the rest of the page furniture already creates a nice margin around the text, the extra lines add noise imo. I'd push for not having any elevation here, what other patterns are offended by this?

File: packages/apps/plugins/plugin-graph/src/graph/types.ts (hunk 2 above)
Comment (zhenyasav) [reply packages/apps/plugins/plugin-graph/src/graph/types.ts:102]: sorry typo as I was messing around understanding this

File: packages/sdk/vault/vault.html (hunk 3 above)
Comment (zhenyasav) [reply packages/sdk/vault/vault.html:37]: typo, thx

**Comment by zhenyasav** [comment]
> I think I'd prefer to take a path that doesn’t trade the main area’s usability / expressive range for the topbar’s visual appeal, which is the only reason I'm requesting changes. I'm looking forward to trying out all the other ideas here!

we should discuss this in more detail, it's unclear to me what the concrete position is here (e.g. usability/expressive range?)

**Comment by richburdon** [comment]
Looking good. Some minor comments:

1. I assume the "flashing circle" when clicking is an artifact of how you're recording this?
2. The scroll-under header opacity is clean in this demo, but I think it will work well with other plugins (e.g., Table plugin) and think we should instead go with a Notion-like header bar.
3. I would really like to see larger dedicated buttons to create a space and create objects (i.e., with a drop down that just lists object types).
4. I feel strongly that the main surface area for the text document (or any other input) should be #fff.
5. Markdown => Documents and use the Folder icon for the folder.
6. Mobile.

**Comment by jessmartin** [comment]
> The scroll-under header opacity is clean in this demo, but I think it will work well with other plugins (e.g., the Table plugin, which also scrolls and the Sketch plugin, which doesn't) and think we should instead go with a Notion-like header bar.

I agree with @richburdon on this one. I appreciate that you're trying to explore a distinct design direction here with the glass scroll-under look and I understand the priority of not chewing up vertical space with a fixed element, but I think a fixed header provides the most flexibility and control for plugins. 

Here's some use cases I'm thinking of:

**Some plugins will want additional "menu" or controls in their main area** and the scroll under will just look weird. For example, the TLDraw plugin:
LIGHT:
image
DARK:
image

In cases like these, how do we create a visual separation between the header and plugin's UI such that they don't clash?

**Some plugins don't scroll, or scroll _within_ the main area in creative ways.** For example, Kanban rows won't scroll up, the scrollviews are within each column. The table plugin will have fixed headers that won't scroll.
LIGHT:
image
DARK:
image

Same challenge: how to differentiate from the "plugin's area" and the header in a cohesive way.

Some ideas:
- If you're really committed to the "glassy" design direction (we're not yet using that style anywhere else in the UI that I see), you could make the header fixed and also transparent/glassy on scroll-under. When there's not content underneath, the transparency wouldn't take effect.
- Another option would be for plugins in the main area to opt-in to different header styles if more vertical space is desired.
- Yet another option is for the header to "auto-hide" on scroll down and "show" on scroll-up, which frees up the vertical space in cases where plugins are long and scroll-y.

All that being said, it seems to me the quickest thing here is to just "copy Notion" or one of the many examples of fixed header, three-column layouts.

**Comment by jessmartin** [comment]
A few random comments:

- Can we have a dedicated "+" to add item button on Space hover as discussed in Figma?
image
A la Notion:
image
- Can we use the same icon to open/close the two sidebars? I like this icon, but really don't care as long as they are the same on both sides:
image
- I know we just moved the "Powered by DXOS" into the settings, but it actually did a nice job of anchoring the bottom of the left sidebar. Minor vote for moving it back there again. :-P

**Pull request closed without merging**